specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-03-30-preview/automanage.json

```diff
@@ -0,0 +1,421 @@
+            "schema": {
+              "$ref": "#/definitions/ConfigurationProfileAssignmentIntent"
+            }
+          },
+          "200": {
+            "description": "OK. The configuration profile assignment intent for the VM was updated successfully.",
+            "schema": {
+              "$ref": "#/definitions/ConfigurationProfileAssignmentIntent"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": false,
+        "produces": [
+          "application/json"
+        ],
+        "consumes": [
+          "application/json"
+        ]
+      },
+      "get": {
+        "tags": [
+          "ConfigurationProfileAssignmentIntents"
+        ],
+        "operationId": "ConfigurationProfileAssignmentIntents_Get",
+        "description": "Get information about a configuration profile assignment intent",
+        "x-ms-examples": {
+          "Get a configuration profile assignment intent": {
+            "$ref": "./examples/getConfigurationProfileAssignmentIntent.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "configurationProfileAssignmentIntentName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The configuration profile assignment intent name."
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/VmNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The configuration profile assignment intent was returned successfully.",
+            "schema": {
+              "$ref": "#/definitions/ConfigurationProfileAssignmentIntent"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "produces": [
+          "application/json"
+        ],
+        "consumes": [
+          "application/json"
+        ]
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Automanage/configurationProfileAssignmentIntents": {
```

Comment:
You should be able to list the configurationProfileAssignments (and configurationProfileAssignmentIntents) on a single VM as well. Right now the user has to list at the RG level then try to filter down to just the VM they care about to see eveyrthing on one VM.

Reply:
We will have one configurationProfileAssignments (and configurationProfileAssignmentIntents) on a single VM with the name **default**.

We will not multiple configurationProfileAssignments (and configurationProfileAssignmentIntents) on a single VM. So the list  the configurationProfileAssignments (and configurationProfileAssignmentIntents) on a single VM is not required.

Reply:
I hope I am clear on the above reply.

Reply:
I don't quite follow your response. You are saying that you would not have multiple configurationProfileAssignmentIntents but your PUT operation above allows to create multiple because you have this: configurationProfileAssignmentIntentName if your intent is to always have one, then you should replace: /{configurationProfileAssignmentIntentName} with /default to denote a singleton. Also, where's your configurationProfileAssignments definition?

---
In reply to: [603716726](https://github.com/Azure/azure-rest-api-specs/pull/13556#discussion_r603716726) [](ancestors = 603716726)

Reply:
Thank you for this comment! It is valid question. I will reply after discussing with our team.

Reply:
Fixed this! Thank you!